File: app/src/main/java/com/example/dagger_hilt/domain/DatabaseRepository.java
package com.example.dagger_hilt.domain;

import androidx.lifecycle.LiveData;

import com.example.dagger_hilt.data.datasource.database.dao.MovieDao;
import com.example.dagger_hilt.data.datasource.database.entities.MovieEntity;

import java.util.List;

import javax.inject.Inject;

public class DatabaseRepository {

    @Inject
    MovieDao movieDao;

    @Inject
    public DatabaseRepository() {
    }

    public LiveData<List<MovieEntity>> getMovies() {
        return movieDao.getMovies();
    }

    public void updateMovie(int id, boolean check) {
      movieDao.updateMovie(id, check);
    }

}

File: app/src/main/java/com/example/dagger_hilt/domain/DatabaseRepository.kt
package com.example.dagger_hilt.domain

import javax.inject.Inject
import com.example.dagger_hilt.data.datasource.database.dao.MovieDao
import androidx.lifecycle.LiveData
import com.example.dagger_hilt.data.datasource.database.entities.MovieEntity

class DatabaseRepository @Inject constructor(private val movieDao: MovieDao) {
    fun updateMovie(id: Int, check: Boolean) = movieDao.updateMovie(id, check)

    val movies: LiveData<List<MovieEntity>>
        get() = movieDao.getMovies()
}